### ISSUE TITLE
Anthropic API key stop working (repeatedly)

### ISSUE THREAD

**Issue by rb81** [BoltAI/BoltAI#219]
This occurs on macOS (latest). I'll be able to use Anthropic fine for quite some time, then suddenly start getting `status code 401invalid x-api-key`. When checking the settings, the API key seems to still be there, but now there's an error when saving (see attachment).

![Screenshot 2024-12-02 at 11 00 23 AM](https://github.com/user-attachments/assets/b64b84ec-df4c-4737-a9e3-d17c882f59d4)

**Comment by longseespace** [comment]
Can you click the "eye" button to see if the API key is correct. I can't seem to reproduce from my end. But there might be an issue with API key encryption/decryption.

**Comment by rb81** [comment]
@longseespace - Yes, checked the API key and it is correct. I've even tried changing the key. First time works perfectly, then after a while (perhaps after a restart) the key no longer works. I then have to change the key and it works again. Key is active and valid in Anthropic Console.

**Comment by longseespace** [comment]
I see. Does it happen after app restart, or OS restart?

**Comment by longseespace** [comment]
Happy to jump to a call to help. Also please join our Discord server for faster support.

**Comment by rb81** [comment]
@longseespace - Okay, seems the problem is with my custom folder location. After restarting my computer, there may be an issue with Bolt loading settings from the custom folder immediately, so it resorts to the default location. I didn't notice this at first. If I've set a custom folder for Bolt, I would expect it to always use that folder. If it doesn't find the location, it should throw an error or prompt the user to use the default location in my opinion.

**Comment by rb81** [comment]
Note: My custom folder location is on an external device, which is why there may be an issue loading the custom folder after a restart. Might take a second for the device to remount and be accessible.

**Comment by longseespace** [comment]
> @longseespace - Okay, seems the problem is with my custom folder location. After restarting my computer, there may be an issue with Bolt loading settings from the custom folder immediately, so it resorts to the default location. I didn't notice this at first. If I've set a custom folder for Bolt, I would expect it to always use that folder. If it doesn't find the location, it should throw an error or prompt the user to use the default location in my opinion.

Interesting. No, BoltAI doesn't load the default DB when it couldn't load the one in your folder. If that happens, BoltAI would just crash. In this case, I think it does connect to the cached version of the db file on your machine. This is why it's generally not recommended to use an external folder.

Trying my best to get the Cloud Sync to work. Sorry.

**Comment by rb81** [comment]
@longseespace - It's definitely resorting to a different folder. I get different chat messages depending on whether my default location is mounted or not. Furthermore, the API key issue seems to be resolved when mounting the device and reloading Bolt. Another interesting point is the API key issue only relates to Anthropic. OpenAI, Ollama, and Gemini are all fine regardless of whether the custom folder location is mounted or not.

**Comment by rb81** [comment]
@longseespace - In the same way the Photos app (or many other apps) prompt the user if the custom folder location is not found, Bolt can simply offer the user to select a new location (or simply exit) if not found.

**Comment by longseespace** [comment]
@rb81 Yeah that would be the best UX. Though I'm not quite sure it loads the default. I will do some more testing hmmm. Can you share your setup? Which cloud service are you using? I think this would not happen if the folder is not a cloud folder (mounting is not needed)

**Comment by rb81** [comment]
@longseespace - On another note, why is there a cache of the DB being stored outside the custom folder location? Defeats the purpose, doesn't it?

**Comment by rb81** [comment]
> @rb81 Yeah that would be the best UX. Though I'm not quite sure it loads the default. I will do some more testing hmmm. Can you share your setup? Which cloud service are you using? I think this would not happen if the folder is not a cloud folder (mounting is not needed)

It's not a cloud service. I have an external drive connected to a USB hub. I have Jettison manage the mounting/unmounting. I've set the custom folder location to a folder on this drive.
